lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -0,0 +1,527 @@
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_target_group(connection, module):
+
+    try:
+        return connection.describe_target_groups(Names=[module.params.get("name")])['TargetGroups'][0]
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'TargetGroupNotFound':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+def diff_list(a, b):
+    """Find the entries in list a that are not in list b"""
+    b = set(b)
+    return [aa for aa in a if aa not in b]
+
+
+def wait_for_status(connection, module, target_group_arn, targets, status):
+    polling_increment_secs = 5
+    max_retries = (module.params.get('wait_timeout') / polling_increment_secs)
+    status_achieved = False
+
+    for x in range(0, max_retries):
+        try:
+            response = connection.describe_target_health(TargetGroupArn=target_group_arn, Targets=targets)
+            if response['TargetHealthDescriptions'][0]['TargetHealth']['State'] == status:
+                status_achieved = True
+                break
+            else:
+                time.sleep(polling_increment_secs)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    result = response
+    return status_achieved, result
+
+
+def create_or_update_target_group(connection, module):
+
+    changed = False
+    params = dict()
+    params['Name'] = module.params.get("name")
+    params['Protocol'] = module.params.get("protocol").upper()
+    params['Port'] = module.params.get("port")
+    params['VpcId'] = module.params.get("vpc_id")
+
+    # If health check path not None, set health check attributes
+    if module.params.get("health_check_path") is not None:
+        params['HealthCheckPath'] = module.params.get("health_check_path")
+
+        if module.params.get("health_check_protocol") is not None:
+            params['HealthCheckProtocol'] = module.params.get("health_check_protocol").upper()
+
+        if module.params.get("health_check_port") is not None:
+            params['HealthCheckPort'] = str(module.params.get("health_check_port"))
+
+        if module.params.get("health_check_interval") is not None:
+            params['HealthCheckIntervalSeconds'] = module.params.get("health_check_interval")
+
+        if module.params.get("health_check_timeout") is not None:
+            params['HealthCheckTimeoutSeconds'] = module.params.get("health_check_timeout")
+
+        if module.params.get("healthy_threshold_count") is not None:
+            params['HealthyThresholdCount'] = module.params.get("healthy_threshold_count")
+
+        if module.params.get("unhealthy_threshold_count") is not None:
+            params['UnhealthyThresholdCount'] = module.params.get("unhealthy_threshold_count")
+
+        if module.params.get("successful_response_codes") is not None:
```

Comment:
If this is every None then line [341](https://github.com/ansible/ansible/pull/19492/files#diff-c008972f0ccccd4a4d9b8c1de95c4d08R341) is going to raise an ugly error. Since this defaults to 200 I think you should remove the check here, since if they set it to an empty string it will cause problems anyway (and they should probably know better).